src/dcap_provider.cpp

```diff
@@ -1004,3 +1140,386 @@ extern "C" void sgx_free_qe_identity_info(
+    const uint8_t* fmspc,
+    const uint16_t fmspc_size,
+    const char* pck_ca,
+    sgx_ql_qve_collateral_t** pp_quote_collateral)
+{
+    log(SGX_QL_LOG_INFO, "Getting quote verification collateral");
+    sgx_ql_qve_collateral_t* p_quote_collateral = nullptr;
+
+    try
+    {
+        if (fmspc == nullptr)
+        {
+            log(SGX_QL_LOG_ERROR, "FMSPC is null");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        if (fmspc_size == 0)
+        {
+            log(SGX_QL_LOG_ERROR, "FMSPC buffer size is 0");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        if (pck_ca == nullptr)
+        {
+            log(SGX_QL_LOG_ERROR, "PCK CA is null");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        if (*pp_quote_collateral != nullptr)
+        {
+            log(SGX_QL_LOG_ERROR,
+				"Collateral pointer is not null. This memory will be allocated by "
+				"this library");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        std::string requested_ca;
+        if (strcmp(CRL_CA_PROCESSOR, pck_ca) == 0)
+        {
+            requested_ca = PROCESSOR_CRL_NAME;
+        }
+
+        if (strcmp(CRL_CA_PLATFORM, pck_ca) == 0)
+        {
+            log(SGX_QL_LOG_ERROR, "Platform CA CRL is not supported");
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        if (requested_ca.empty())
+        {
+            log(SGX_QL_LOG_ERROR,
+                "PCK CA must be either %s or %s",
+                CRL_CA_PROCESSOR,
+                CRL_CA_PLATFORM);
+            return SGX_QL_ERROR_INVALID_PARAMETER;
+        }
+
+        std::string str_fmspc((char*)fmspc, fmspc_size);
+        quote3_error_t operation_result;
+        std::vector<uint8_t> pck_crl;
+        std::string pck_issuer_chain;
+        std::vector<uint8_t> root_ca_crl;
+        std::string root_ca_chain;
+        std::vector<uint8_t> tcb_info;
+        std::string tcb_issuer_chain;
+        std::vector<uint8_t> qe_identity;
+        std::string qe_identity_issuer_chain;
+
+        // Get PCK CRL
+        std::string pck_crl_url = build_pck_crl_url(requested_ca, API_VERSION);
+        log(SGX_QL_LOG_INFO,
+            "Fetching PCK CRL from remote server: '%s'.",
+            pck_crl_url.c_str());
+        operation_result = get_collateral(
+            pck_crl_url, headers::CRL_ISSUER_CHAIN, pck_crl, pck_issuer_chain);
+        if (operation_result != SGX_QL_SUCCESS)
+        {
+            log(SGX_QL_LOG_ERROR, "Error fetching PCK CRL: %d", operation_result);
+            return operation_result;
+        }
+
+        // Get Root CA CRL
+        std::string root_ca_crl_url = build_pck_crl_url(ROOT_CRL_NAME, API_VERSION);
+        log(SGX_QL_LOG_INFO,
+            "Fetching Root CA CRL from remote server: '%s'.",
+            root_ca_crl_url.c_str());
+       operation_result = get_collateral(
+			root_ca_crl_url, headers::CRL_ISSUER_CHAIN, root_ca_crl, root_ca_chain);

```

Comment:
>			 [](start = 0, length = 3)

Minor - Replace TABs with spaces. Please check globally.

Reply:
Thanks! I've modified my editor settings and checked the files included in this pull request